### DISCUSSION TITLE
How to use JS modules?

### DISCUSSION THREAD

**Discussion by LiveNathan** [wimdeblauwe/ttcli#83] [Q&A] [answered]
To use javascript modules do we need to install a bundler? 
I tried copying over some JS from another project and it's throwing a lot of errors. It acts like it can't find any of the files, which are all imported at the top of main.js.

![2023-11-09 at 11 38 11@2x](https://github.com/wimdeblauwe/ttcli/assets/60303904/85f4038a-0b33-4fe8-9ef2-9f28191ebd67)

![2023-11-09 at 11 38 37@2x](https://github.com/wimdeblauwe/ttcli/assets/60303904/ba3f0054-fb5e-4bd5-b20c-e6c52f49231d)

**Comment by wimdeblauwe** [comment]
I don't understand the question I am afraid. I also don't understand what ttcli has to do with this. Maybe you can ask your question on stackoverflow?

**Reply by LiveNathan** [reply]
Thanks Wim. The reason I thought this might be a ttcli related issue is that it currently does not allow us to use JavaScript modules. It seems like it needs a bundler like Rollup. 

So maybe it's actually a feature request?

**Reply by wimdeblauwe** [reply]
I have not worked with Rollup before. As it is not something I need, I will personally not spend time on it. But if you or somebody else creates a PR for it, then I am willing to review it.

**Reply by LiveNathan** [reply, marked as answer]
Hey Wim, I created an [MRE](https://github.com/LiveNathan/spring-boot-js-module-demo) today in order to demonstrate that js modules do not work, but it worked immediately so the problem must be somewhere else in my project.
